lib/bracket-matcher.coffee

```diff
@@ -81,7 +82,7 @@ class BracketMatcher
       @editor.moveRight()
       false
     else if autoCompleteOpeningBracket
-      @editor.insertText(text + pair)
+      @origEditorInsertText(text + pair)
       @editor.moveLeft()
```

Comment:
Hi. `@origEditorInsertText` is the same as `@editor.insertText` was before we did `_.adviseBefore` on it. 

The recursion is that `BracketMatcher::insertText` calls `BracketMatcher::insertText` again when it calls `@editor.insertText`, then decides nontrivially that it doesn't have to do anything and returns `true` which causes `adviseBefore` to finally call the original `TextEditor::insertText`.